### PR TITLE
[PATCH] Fix from Francois Killeen to power module tests to allow unsu…

### PR DESCRIPTION
**Pull request by jakegwood**
…pported power mode and fix EnterIdleCancel test.

Description
-----------

Checklist:
----------

- [ x ] I have tested my changes. No regression in existing tests.
- [ x ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/common_io/test/test_iot_power.c [hunk 1]
```diff
@@ -368,12 +375,12 @@ TEST( TEST_IOT_POWER, AFQP_IotPower_EnterIdle )
     if( lRetVal != IOT_POWER_FUNCTION_NOT_SUPPORTED )
     {
         TEST_ASSERT_EQUAL( IOT_POWER_SUCCESS, lRetVal );
-    }
 
-    /* Read the performance mode and verify high performance mode */
-    lRetVal = iot_power_get_mode( xPowerHandle, &xPowerMode );
-    TEST_ASSERT_EQUAL( IOT_POWER_SUCCESS, lRetVal );
-    TEST_ASSERT_EQUAL( eHighPowerPerfMode, xPowerMode );
+        /* Read the performance mode and verify high performance mode */
```
Comment (vjjunnut) [inline libraries/abstractions/common_io/test/test_iot_power.c:379]: This should be in else condition if you are fixing it. As if set_mode returns NOT_SUPPORTED, then we dont need to check for the mode.

File: libraries/abstractions/common_io/test/test_iot_power.c [hunk 2]
```diff
@@ -445,12 +452,24 @@ TEST( TEST_IOT_POWER, AFQP_IotPower_EnterIdleCancel )
     if( lRetVal != IOT_POWER_FUNCTION_NOT_SUPPORTED )
     {
         TEST_ASSERT_EQUAL( IOT_POWER_SUCCESS, lRetVal );
+
+        /* Read the performance mode and verify high performance mode */
```
Comment (vjjunnut) [inline libraries/abstractions/common_io/test/test_iot_power.c:456]: Same here - should be in else.

File: libraries/abstractions/common_io/test/test_iot_power.c [hunk 3]
```diff
@@ -445,12 +452,24 @@ TEST( TEST_IOT_POWER, AFQP_IotPower_EnterIdleCancel )
...
+      /* Set the power mode to normal mode */
+      lRetVal = iot_power_set_mode( xPowerHandle, eNormalPowerPerfMode );
 
-    /* Read the performance mode and verify high performance mode */
-    lRetVal = iot_power_get_mode( xPowerHandle, &xPowerMode );
-    TEST_ASSERT_EQUAL( IOT_POWER_SUCCESS, lRetVal );
-    TEST_ASSERT_EQUAL( eHighPowerPerfMode, xPowerMode );
+      TEST_ASSERT_EQUAL( IOT_POWER_SUCCESS, lRetVal );
+
+      /* Read the performance mode and verify normal performance mode */
```
Comment (vjjunnut) [inline libraries/abstractions/common_io/test/test_iot_power.c:468]: Here as well - should be in else.

File: libraries/abstractions/common_io/test/test_iot_power.c [hunk 4]
```diff
@@ -779,12 +797,12 @@ TEST( TEST_IOT_POWER, AFQP_IotPower_PCWakeThresholdTest )
...
     {
         TEST_ASSERT_EQUAL( IOT_POWER_SUCCESS, lRetVal );
-    }
 
-    /* Read the performance mode and verify high performance mode */
-    lRetVal = iot_power_get_mode( xPowerHandle, &xPowerMode );
-    TEST_ASSERT_EQUAL( IOT_POWER_SUCCESS, lRetVal );
-    TEST_ASSERT_EQUAL( eHighPowerPerfMode, xPowerMode );
+        /* Read the performance mode and verify high performance mode */
+        lRetVal = iot_power_get_mode( xPowerHandle, &xPowerMode );
```
Comment (vjjunnut) [inline libraries/abstractions/common_io/test/test_iot_power.c:802]: Move this to else condition as well

File: libraries/abstractions/common_io/test/test_iot_power.c [hunk 5]
```diff
@@ -921,12 +939,12 @@ TEST( TEST_IOT_POWER, AFQP_IotPower_ClkSrcOffWakeThresholdTest )
     if( lRetVal != IOT_POWER_FUNCTION_NOT_SUPPORTED )
     {
         TEST_ASSERT_EQUAL( IOT_POWER_SUCCESS, lRetVal );
-    }
 
-    /* Read the performance mode and verify high performance mode */
-    lRetVal = iot_power_get_mode( xPowerHandle, &xPowerMode );
-    TEST_ASSERT_EQUAL( IOT_POWER_SUCCESS, lRetVal );
-    TEST_ASSERT_EQUAL( eHighPowerPerfMode, xPowerMode );
+        /* Read the performance mode and verify high performance mode */
```
Comment (vjjunnut) [inline libraries/abstractions/common_io/test/test_iot_power.c:943]: Move this to else.

File: libraries/abstractions/common_io/test/test_iot_power.c (hunk 1 above)
Comment (fkilleen) [reply libraries/abstractions/common_io/test/test_iot_power.c:379]: I'm not sure I understand..
If `set_mode` returns NOT_SUPPORTED as you mentioned.
Then, `lRetVal` will be equal to `NOT_SUPPORTED`.

This `if` condition checks if `lRetVal != IOT_POWER_FUNCTION_NOT_SUPPORTED` before executing the code.
In this scenario, where high performance mode is not supported, we won't check for the mode.

Comment (vjjunnut) [reply libraries/abstractions/common_io/test/test_iot_power.c:379]: nm. You are correct.

**Review by vjjunnut — DISMISSED** [review]

**Comment by lundinc2** [comment]
I have updated your branch with an uncrustified version of the changes to pass the style check. Will merge after the checks pass.

**Review by lundinc2 — APPROVED** [review]

**Review by cobusve — APPROVED** [review]

**Pull request merged**